core/src/main/java/com/netflix/iceberg/GenericDataFile.java

```diff
@@ -402,6 +421,7 @@ public String toString() {
         .add("null_value_counts", nullValueCounts)
         .add("lower_bounds", lowerBounds)
         .add("upper_bounds", upperBounds)
+        .add("key_metadata", keyMetadata == null ? "null" : "<redacted>")
```

Comment:
Nit: if you're committing another version, then please use `(redacted)` instead because `<` and `>` must be escaped when displayed in web pages. Not a big deal, I just try to stick with parens.